crates/solver/src/liquidity/order_converter.rs

```diff
@@ -71,14 +74,32 @@ struct OrderSettlementHandler {
     scaled_unsubsidized_fee_amount: U256,
 }
 
+/// Returns (`sell_amount`, `fee_amount`) for the given order and adjusts the values accordingly
+/// for limit orders.
+fn compute_synthetic_order_amounts_if_limit_order(order: &Order) -> Result<(U256, U256)> {
+    if order.metadata.class == OrderClass::Limit {
+        // adjust amounts to account for `surplus_fee`
+        let sell_amount = order
+            .data
+            .sell_amount
+            .checked_add(order.data.fee_amount)
+            .context("surplus_fee adjustment would overflow sell_amount")?
+            .checked_sub(order.metadata.surplus_fee)
+            .context("surplus_fee adjustment would underflow sell_amount")?;
+        return Ok((sell_amount, order.metadata.surplus_fee));
```

Comment:
It makes sense that surplus_fee is subtracted from sell_amount. But why is fee_amount added? Maybe was explained in the discussion on Github but when I read this code here it is unclear.
Naively I'd think that surplus_fee already takes into account that the order was a real fee too.

Reply:
Basically, `surplus_fee` is the exact fee amount that needs to be taken for the order for that auction (regardless of what the actually set `fee_amount` is). This leads to the following system of equations:

```
(1)
synthetic_order.fee_amount = surplus_fee

(2)
synthetic_order.sell_amount + synthetic_order.fee_amount
  = order.sell_amount + order.fee_amount
  
Solving for synthetic_order.sell_amount
(3)
synthetic_order.sell_amount
  = order.sell_amount + order.fee_amount - surplus_fee
```

For limit orders, you can imagine they are an order without pro-rated fee. Because the settlement contract always transfers in `order.sell_amount + order.fee_amount`, we need to account for non-0 fee amounts here.